### ISSUE TITLE
`qtmoc` never finished

### ISSUE THREAD

**Issue by arknable** [therecipe/qt#263]
Hi,

This maybe related to https://github.com/therecipe/qt/issues/255, but I created new issue because probably the cause is not protobuf import. Latest fix works on simple project but somehow `qtmoc` never finished on my project. Attached is log messages when running `qtmoc` for half hour, could you please see if you can spot the problem from the log ?

Thanks

[log-qtmoc.txt](https://github.com/therecipe/qt/files/881412/log-qtmoc.txt)

**Comment by therecipe** [comment]
Hey

I will look into it.

**Comment by arknable** [comment]
Thanks a lot ya

**Comment by therecipe** [comment]
It's probably fixed with: https://github.com/therecipe/qt/commit/256e4454bdaf098d34334d3475a52d9b2e95f100

**Comment by arknable** [comment]
Hey, 

I tried and got panic error, attached is log messages.

Thanks

[log.txt](https://github.com/therecipe/qt/files/884988/log.txt)

**Comment by therecipe** [comment]
@arknable 

It seems like you used an version of qtmoc which is at least 14 days old.

Maybe try to run `qtsetup update` to update the tools or if this won't work,
you probably should upgrade the whole repo with: `qtsetup upgrade && qtsetup`

**Comment by arknable** [comment]
ouw, that's weird, I did update the whole repo. let me try again, probably need to remove everything first then pull everything from fresh.

**Comment by bclermont** [comment]
And?

**Comment by arknable** [comment]
@therecipe With fresh repository I'm still getting that panic error. I'm trying `qtsetup upgrade` ... but it's really slow right now.

**Comment by arknable** [comment]
@therecipe `qtsetup upgrade` didn't work. how to check the version of `qtmoc` ? I can't see any `-version` argument on help section.

**Comment by bclermont** [comment]
Maybe you have multiple installed binaries? 

`find / type f -name qtmoc`

**Comment by arknable** [comment]
@bclermont no, I checked.
```
find: /dev/fd/3: Not a directory
find: /dev/fd/4: Not a directory
/xxx/bin/qtmoc
/xxx/src/github.com/therecipe/qt/cmd/qtmoc
/yyy/Downloads/qt-master/cmd/qtmoc
/usr/local/bin/qtmoc
find: type: No such file or directory
find: f: No such file or directory
```

**Comment by arknable** [comment]
`/usr/local/bin/qtmoc` is symlink to real `qtmoc` in `GOPATH` which doesn't exists now because I'm trying to rebuild the binding again.

**Comment by arknable** [comment]
oh sorry, just realized the rebuild has finished, hence there is `/xxx/bin/qtmoc`

**Comment by arknable** [comment]
just tried again, still the same panic. but this only occurs on cross-package import, it doesn't occurred when I copied all cross-package into `main` package, which become the same package as `main.go`.

**Comment by arknable** [comment]
@therecipe This is the last log from my latest try.

[log.txt](https://github.com/therecipe/qt/files/888217/log.txt)

**Comment by bclermont** [comment]
@therecipe I can make it works now. but when there is more value in `GOPATH` it create import that append the other value of `GOPATH` and not the package name

**Comment by arknable** [comment]
@therecipe I tried latest fix and got these errors:
```
# _/xxx/QtTypeTest
./moc.go:57: undefined: custom_qt in custom_qt.QDuration
./moc.go:68: undefined: custom_qt in custom_qt.QDuration_ITF
./moc.go:82: undefined: custom_qt in custom_qt.QDuration
./moc.go:107: undefined: custom_qt in custom_qt.QDuration_ITF
./moc.go:113: undefined: custom_qt in custom_qt.QIP
./moc.go:124: undefined: custom_qt in custom_qt.QIP_ITF
./moc.go:138: undefined: custom_qt in custom_qt.QIP
./moc.go:163: undefined: custom_qt in custom_qt.QIP_ITF
./moc.go:169: undefined: custom_qt in custom_qt.QIPAddr
./moc.go:180: undefined: custom_qt
./moc.go:180: too many errors
```

**Comment by arknable** [comment]
This is fixed, thanks.
